src/command/archive/get/get.c

```diff
@@ -200,74 +362,149 @@ archiveGetCheck(const StringList *archiveRequestList)
+
+            TRY_BEGIN()
+            {
+                // Get the repo storage in case it is remote and encryption settings need to be pulled down
+                storageRepoIdx(repoIdx);
 
-        // Attempt to load the archive info file
-        InfoArchive *info = infoArchiveLoadFile(
-            storageRepo(), INFO_ARCHIVE_PATH_FILE_STR, result.cipherType, cfgOptionStrNull(cfgOptRepoCipherPass));
+                ArchiveGetFindCacheRepo cacheRepo =
+                {
+                    .repoIdx = repoIdx,
+                    .cipherType = cipherType(cfgOptionIdxStr(cfgOptRepoCipherType, repoIdx)),
+                    .archiveList = lstNewP(sizeof(ArchiveGetFindCacheArchive)),
+                    .warnList = strLstNew(),
+                };
+
+                // Attempt to load the archive info file
+                InfoArchive *info = infoArchiveLoadFile(
+                    storageRepoIdx(repoIdx), INFO_ARCHIVE_PATH_FILE_STR, cacheRepo.cipherType,
+                    cfgOptionIdxStrNull(cfgOptRepoCipherPass, repoIdx));
+
+                MEM_CONTEXT_BEGIN(lstMemContext(result.archiveFileMapList))
+                {
+                    cacheRepo.cipherPassArchive = strDup(infoArchiveCipherPass(info));
+                }
+                MEM_CONTEXT_END();
 
-        // Loop through the pg history and determine which archiveId to use based on the first file in the list
-        bool found = false;
-        const String *archiveId = NULL;
-        List *cache = NULL;
+                // Loop through pg history and determine which archiveIds to use
+                StringList *archivePathList = NULL;
 
-        for (unsigned int pgIdx = 0; pgIdx < infoPgDataTotal(infoArchivePg(info)); pgIdx++)
-        {
-            InfoPgData pgData = infoPgData(infoArchivePg(info), pgIdx);
+                for (unsigned int pgIdx = 0; pgIdx < infoPgDataTotal(infoArchivePg(info)); pgIdx++)
+                {
+                    InfoPgData pgData = infoPgData(infoArchivePg(info), pgIdx);
 
-            // Only use the archive id if it matches the current cluster
-            if (pgData.systemId == controlInfo.systemId && pgData.version == controlInfo.version)
-            {
-                archiveId = infoPgArchiveId(infoArchivePg(info), pgIdx);
-                cache = lstNewP(sizeof(ArchiveGetFindCache), .comparator = lstComparatorStr);
+                    // Only use the archive id if it matches the current cluster
+                    if (pgData.systemId == controlInfo.systemId && pgData.version == controlInfo.version)
+                    {
+                        const String *archiveId = infoPgArchiveId(infoArchivePg(info), pgIdx);
+                        bool found = true;
 
-                found = archiveGetFind(
-                    strLstGet(archiveRequestList, 0), archiveId, &result, cache, strLstSize(archiveRequestList) == 1);
+                        // If the archiveId is in the past make sure it has some files
```

Comment:
True -- but since we delete the dir in expire when all the files have been expired this should work, right? So I just need to update the comment to be more accurate?

Reply:
yes

Reply:
Fixed the comment in d4a0998.